src/mempool/mempool.rs

```diff
@@ -62,29 +69,62 @@ impl Mempool {
         }
     }
 
-    fn message_exists_in_trie(&mut self, fid: u64, trie_key: Vec<u8>) -> bool {
+    fn message_already_exists(&mut self, message: &MempoolMessage) -> bool {
+        let fid = message.fid();
         let shard = self.message_router.route_message(fid, self.num_shards);
         let stores = self.shard_stores.get_mut(&shard);
+        // Default to false in the orror paths
         match stores {
             None => {
                 error!("Error finding store for shard: {}", shard);
                 false
             }
-            Some(stores) => {
-                // TODO(aditi): The engine reloads its ref to the trie on commit but we maintain a separate ref to the trie here.
-                stores.trie.reload(&stores.db).unwrap();
-                match stores.trie.exists(
-                    &merkle_trie::Context::new(),
-                    &stores.db,
-                    trie_key.as_ref(),
-                ) {
-                    Err(err) => {
-                        error!("Error finding key in trie: {}", err);
-                        false
+            Some(stores) => match message {
+                MempoolMessage::UserMessage(message) => match &message.data {
+                    None => false,
+                    Some(message_data) => {
+                        let ts_hash = make_ts_hash(message_data.timestamp, &message.hash).unwrap();
+                        let set_postfix = type_to_set_postfix(message_data.r#type());
+                        let primary_key =
+                            make_message_primary_key(fid, set_postfix as u8, Some(&ts_hash));
+                        let existing_message = get_message_by_key(
```

Comment:
This is too low level for the mempool. It should just be doing `stores.message_exists(&message);`